src/Servers/Kestrel/Core/src/Internal/HttpsConnectionMiddleware.cs

```diff
@@ -69,33 +71,58 @@ public HttpsConnectionAdapter(HttpsConnectionAdapterOptions options, ILoggerFact
             }
 
             _options = options;
-            _logger = loggerFactory?.CreateLogger<HttpsConnectionAdapter>() ?? (ILogger)NullLogger.Instance;
+            _logger = loggerFactory?.CreateLogger<HttpsConnectionMiddleware>();
         }
-
-        public bool IsHttps => true;
-
-        public Task<IAdaptedConnection> OnConnectionAsync(ConnectionAdapterContext context)
+        public Task OnConnectionAsync(ConnectionContext context)
         {
-            // Don't trust SslStream not to block.
             return Task.Run(() => InnerOnConnectionAsync(context));
         }
 
-        private async Task<IAdaptedConnection> InnerOnConnectionAsync(ConnectionAdapterContext context)
+        private async Task InnerOnConnectionAsync(ConnectionContext context)
         {
             SslStream sslStream;
             bool certificateRequired;
             var feature = new Core.Internal.TlsConnectionFeature();
             context.Features.Set<ITlsConnectionFeature>(feature);
             context.Features.Set<ITlsHandshakeFeature>(feature);
 
+            // TODO: Handle the cases where this can be null
```

Comment:
Nit: remove TODO or do it.

Reply:
I'll tackle them across the code base in a separate PR. 

PS: https://github.com/aspnet/AspNetCore/blob/0648edf7c2c481309272697c0e0b60e47696f9f7/src/Servers/Kestrel/Transport.Libuv/src/Internal/Networking/UvStreamHandle.cs#L47

Reply:
I'm aware there are TODO comments in Kestrel. That one was checked in [five years ago](https://github.com/aspnet/AspNetCore/commit/de6c32dc4b33c066bad84e65b76e05e46202769a) when Kestrel was still a test server. I bet you could find a bunch more recent examples than that, but I don't really see the point. I'm glad you don't plan to just leave this there though.